packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -342,7 +342,104 @@ describe('plugin-devices', () => {
       });
     });
 
-    describe('#getWebSocketUrl()', () => {});
+    describe('#getWebSocketUrl()', () => {
+      let services;
+
+      beforeEach('destructure services', () => {
+        services = webex.internal.services;
+      });
+
+      describe('when wait is truthy', () => {
+        let wait;
+
+        beforeEach('set wait', () => {
+          wait = true;
+        });
+
+        describe('when the device is registered', () => {
+          beforeEach('register the device', () => devices.register());
+
+          it('should resolve the promise with the websocket url',
+            () => devices.getWebSocketUrl(wait)
+              .then((url) => {
+                assert.isDefined(url);
+                assert.isTrue(services.isServiceUrl(url));
+                assert.include(url, 'mercury');
+              }));
+        });
+
+        describe('when the device is not registered', () => {
+          describe('when the device successfully registers', () => {
+            beforeEach(() => {
+              // This needs to not be returned, as it must be pending during the
+              // it clause.
+              devices.register();
+            });
+
+            it('should resolve the promise with the websocket url', () => {
+              assert.isFalse(devices.registered);
+
+              return devices.getWebSocketUrl(wait)
+                .then((url) => {
+                  assert.isDefined(url);
+                  assert.isTrue(services.isServiceUrl(url));
+                  assert.include(url, 'mercury');
+                });
+            });
+          });
+
+          describe('when the device never registers', () => {
```

Comment:
This describe block seems extraneous. "when the device is not registered, it should return a rejected promise" works.

Reply:
I believe that is what it currently does.
```js
describe('when the device never registers', () => {
  it('should return a rejected promise', () => {
    assert.isFalse(devices.registered); // assert that it never registered
    assert.isRejected(devices.getWebSocketUrl(wait)); // assert that it rejected
  });
});
```
Let me know if you have an example or more info on how I could improve this.

Reply:
You can leave the `it` but remove the entire describe block of `describe('when the device never registers', () => {`. The surrounding describe block is sufficient.

Reply:
I see, made this change.